### PR TITLE
Solutions part2

### PR DESCRIPTION
**Pull request by nataliadombrowska**
# Problem descriptions

## Pair Summing

Write a function that counts how many pairs of numbers in an array add up to a given number. To follow up: how to do this more efficiently.

## Letters of Alphabet
Implement a function to determine which letters of the alphabet were missing from an input string.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (daustin-scottlogic) [inline .gitignore:1]: Would be handy if you could add the `.iml` and `.class` files to the gitignore, and remove them from git

File: src/LettersOfAlphabet.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import org.junit.Test;
```
Comment (daustin-scottlogic) [inline src/LettersOfAlphabet.java:1]: Maybe start adding in the questions as comments at the top of your files, just so its clear what they're answering

File: src/LettersOfAlphabet.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    public List<Character> findMissingLetters(String word) {
+        List<Character> missingLetters = new ArrayList<>();
+        List<Character> alphabet = Arrays.asList(
+                'a','b','c','d','e','f','g','h','i','j',
+                'k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'
+        );
+
+        //convert the string to a list of characters
```
Comment (daustin-scottlogic) [inline src/LettersOfAlphabet.java:22]: have you considered `.toCharArray()` ?

File: src/LettersOfAlphabet.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+        for(int i = 0; i < word.length(); i++){
+            letters[i] = lowerCaseWord.charAt(i);
+        }
+
+        List<Character> wordAsList = Arrays.asList(letters);
+
+        //Loop through the alphabet and add any missing letters
+        for(int i = 0; i < alphabet.size(); i++) {
+            if(!(wordAsList.contains(alphabet.get(i)))){
```
Comment (daustin-scottlogic) [inline src/LettersOfAlphabet.java:34]: you can probably ditch 1 pair of brackets
```suggestion
            if(!wordAsList.contains(alphabet.get(i))){
```

File: src/LettersOfAlphabet.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                missingLetters.add(alphabet.get(i));
```
Comment (daustin-scottlogic) [inline src/LettersOfAlphabet.java:35]: consider a `char letter` variable so you don't have to `get` twice

File: src/SingleCharacterSearch.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+
+import java.util.HashMap;
+
+import static java.lang.invoke.MethodHandles.throwException;
+import static org.junit.Assert.*;
+
+public class SingleCharacterSearch {
```
Comment (daustin-scottlogic) [inline src/SingleCharacterSearch.java:9]: Is this one ready for review? Question isn't in the PR description

File: src/PairSumming.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class PairSumming {
+    public static void main(String[] args) {
+        JUnitCore.main("PairSumming");
+    }
+
+    public int pairSumming(Integer[] arr, int target) {
```
Comment (daustin-scottlogic) [inline src/PairSumming.java:15]: This looks like it does the job to me. Might be worth finding out if you need to be able to solve these without things like Lists.

File: src/SingleCharacterSearch.java (hunk 6 above)
Comment (nataliadombrowska) [reply src/SingleCharacterSearch.java:9]: It got added because because i changed the single search approach.

File: src/LettersOfAlphabet.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        List<Character> expected = Arrays.asList(arr);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void test4() {
+        List<Character> actual = findMissingLetters("bcdefghijklmnopqrstuvwxyz");
+        Character[] arr = {'a'};
+        List<Character> expected = Arrays.asList(arr);
+        assertEquals(expected, actual);
```
Comment (jluxford-scottlogic) [inline src/LettersOfAlphabet.java:72]: This looks to be the same as `test1` (though I might be missing something), was this suppose to be testing something different?

File: src/PairSumming.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        List<Integer> numbers = new LinkedList<Integer>(Arrays.asList(arr));
+
+        for(int i = 0; i < arr.length - 1; i++) {
+            int difference = target - arr[i];
+            numbers.remove(arr[i]);
+            if(numbers.contains(difference)){
+                count++;
+            }
+        }
+        return count;
```
Comment (jluxford-scottlogic) [inline src/PairSumming.java:26]: How would this behave with a list that looks like `[0, 2, 0, 2, 2]` with target of `2` and is that what you would expect?

**Pull request merged**